exec/cnex/cell.c

```diff
@@ -342,11 +402,12 @@ int32_t cell_compareCell(const Cell * s, const Cell * d)
     }
 
     switch (s->type) {
-        case cString:   return string_compareString(s->string, d->string);
-        case cAddress:  return s->address == d->address != 0;
-        case cBoolean:  return s->boolean == d->boolean != 0;
-        case cArray:    return s->array_size == d->array_size != 0;
-        case cNumber:   return bid128_quiet_equal(s->number, d->number) != 0;
+        case cAddress:      return !(s->address == d->address);
+        case cArray:        return !(s->array_size == d->array_size);
+        case cBoolean:      return !s->boolean == d->boolean;
```

Comment:
This still isn't right. To be correct, this would have to be `!(s->boolean == d->boolean)`. But instead of using a confusing expression like that, why not use `s->boolean != d->boolean`?

Reply:
Changed to all use the != construct instead.